### PR TITLE
Add `Scrollable` and `ScrollBar` widgets

### PR DESCRIPTION
**Pull request by penguinolog**
Original work @rndusr with fixes from @markqvist

* Add basic type annotations
* Add basic set of block symbols
* Add minimal set of tests

* Use protocol-based API compatibility checking.
* Require `typing-extensions` since `Protocol` is part of `typing` from python 3.8.

Fix: #226

##### Checklist
- [X] I've ensured that similar functionality has not already been implemented
- [X] I've ensured that similar functionality has not earlier been proposed and declined
- [X] I've branched off the `master` or `python-dual-support` branch
- [X] I've merged fresh upstream into my branch recently
- [X] I've ran `tox` successfully in local environment
- [X] I've included docstrings and/or documentation and/or examples for my code (if this is a new feature)

### CODE REVIEW

File: urwid/widget/widget.py [hunk 1]
```diff
@@ -284,58 +284,6 @@ class Widget(metaclass=WidgetMeta):
        widget renders the same size regardless of the value of the *focus*
        parameter.
 
-
```
Comment (penguinolog) [inline urwid/widget/widget.py:287]: These parts of docstring duplicates implemented methods and causes warnings during documentation build.

File: urwid/widget/scrollable.py [hunk 2]
```diff
@@ -0,0 +1,593 @@
...
+          - `rows_max` takes `size` and `focus` and returns the total number of rows `widget` can render.
+
+        `thumb_char` is the character used for the scrollbar handle.
+        `trough_char` is used for the space above and below the handle.
+        `side` must be 'left' or 'right'.
+        `width` specifies the number of columns the scrollbar uses.
+        """
+        if Sizing.BOX not in widget.sizing():
+            raise ValueError(f"Not a box widget: {widget!r}")
+        if not isinstance(widget, SupportsScroll):
```
Comment (penguinolog) [inline urwid/widget/scrollable.py:451]: @rndusr with `Protocol` classes we can check for all needed API using `isinstance` check

**Review by wardi — APPROVED** [review]
Looks like a great feature. Might be good to have some direction for users on the pros and cons of choosing to use these new widgets vs ListBox.

File: docs/manual/widgets.rst [hunk 3]
```diff
@@ -578,7 +577,55 @@ When this is defined it will be used by :meth:`ListBox.__iter__` and
...
 .. currentmodule:: urwid
 
+Scrollable Widgets
+==================
+Scrollable widgets can scroll long content which normally not fit into the screen resolution.
+Scrolling is normally supported using keyboard positioning keys and mouse wheel.
+
+Comparing to the :class:`ListBox`,
+:class:`Scrollable` handles fixed and flow widgets directly instead of using list of small widgets.
+And :class:`ScrollBar` provide visual scrollbar with scrolling position information.
```
Comment (wardi) [inline docs/manual/widgets.rst:587]: Can ScrollBar be used with ListBox?

Comment (penguinolog) [reply docs/manual/widgets.rst:587]: Only if `ListBox` will be extended to support positioning API

**Pull request merged**